lib/materials/concrete.rb

```diff
@@ -1,10 +1,35 @@
+require './lib/materials/base'
+
 module Materials
   class Concrete < Base
-    attr_reader :elastic_module, :design_compression_strength
+    attr_reader :elastic_module, :design_compression_strength, :specific_weight
+
+    def self.new_reinfoced_concrete(elastic_module:, design_compression_strength:)
+      concrete = new(elastic_module:, design_compression_strength:)
+      concrete.reinforced_concrete
+
+      concrete
+    end
+
+    def self.new_simple_concrete(elastic_module:, design_compression_strength:)
+      concrete = new(elastic_module:, design_compression_strength:)
+      concrete.simple_concrete
+
+      concrete
+    end
```

Comment:
I'd rather prefer to keep things simple and avoid having all these wrappers around `initialize`. Why not just receive `speicfic_weight` as an argument in `initialize`, and set it there directly?

```ruby
    def initialize(elastic_module:, design_compression_strength:, specific_weight:)
      @elastic_module = elastic_module.to_f
      @design_compression_strength = design_compression_strength.to_f
      @specific_weight = specific_weight.to_f
    end
```

Reply:
yeah, agree!